superset/assets/javascripts/SqlLab/components/VisualizeModal.jsx

```diff
@@ -271,4 +270,19 @@ class VisualizeModal extends React.PureComponent {
 VisualizeModal.propTypes = propTypes;
 VisualizeModal.defaultProps = defaultProps;
 
-export default VisualizeModal;
+function mapStateToProps(state) {
+  return {
+    datasource: state.datasource,
+    errorMessage: state.errorMessage,
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    actions: bindActionCreators(actions, dispatch),
+  };
+}
+
+export { VisualizeModal };
+export default connect(mapStateToProps, mapDispatchToProps)(VisualizeModal);
```

Comment:
About 'When to Introduce Containers?', I feel this article is reasonable: [Presentational and Container Components](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0).

For VisulizeModal component, I feel its state (datasource, and errorMessage) are rarely needed by other components. If we make it a dump component, since VisulizeModal is used by QueryTable and ResultSet, does it means both parents need to pass additional pros down? So I prefer to wrap its own states in a container.

Currently SqlLab code structure is more like:
- components
  - component1.jsx
  - component2.jsx
- actions_for_all_components.js
- reducers_for_all_actions.js
The problem is now presentation components are injected with so many actions it doesn't need. this makes sub-component hard to be reused outside SqlLab.

I will prefer to have it breakdown into
- contaienr1
  - container1.jsx
  - component1.jsx
  - action_for_comp1.js
  - reducer_for_comp1.js
- component2
  - component2.js
  - container2.js
  - action_for_comp2
  - reducer_for_comp2

I am still new to this code base, would like to hear your opinions. Thank you!